File: back-end/classes/userDB.php
<?php 
require_once("dbHandler.php");

class UserDB extends DBHandler{
    private string $username;
    private string $mail;
    private string $password;

    function __construct(string $username,string $mail,string $password){
        parent::__construct();
        $this->mail = $mail;
        $this->username = $username;
        $this->password = password_hash($password,PASSWORD_BCRYPT);
    }

    function addUser(){
        $arrayData = [
            "username" => $this->username,
            "mail"=> $this->mail,
            "password" => $this->password,
        ];
        $idUSer = $this->insert($arrayData,"user");
        return $idUSer;
    }
}


?>

File: connectProject/src/connectProject/UserDB.java
package connectProject;

import java.util.*;

public class UserDB {
	private String username;
	private String mail;
	private String password;
	private DbHandler DB = new DbHandler();
	UserDB(String username,String mail,String password){
		this.password = password;
		this.username = username;
		this.mail = mail;
	}
	
	public int createUser() {
		LinkedHashMap<String,String> user = new LinkedHashMap<String,String>();
		user.put("password", password);
		user.put("mail", mail);
		user.put("username", username);

		int res = DB.insertInDatabase(mail, user);
		return res;
	}
}
